src/android/Print.java

```diff
@@ -722,16 +724,31 @@ private int getPrinterSeries(String target) {
 		return series;
 	}
 
+	// As of version code 95 (version name 1.2.95), the PrintConnect app must be triggered in the foreground
+	private static final long NEW_PRINT_CONNECT_VERSION = 95;
+
 	private void sendToPrintConnect(JSONArray args, final CallbackContext callbackContext) throws JSONException {
 		final String zplString = args.getString(0);
 		byte[] passthroughBytes = null;
+		boolean requiresForeground = false;
 
 		try {
 			passthroughBytes = zplString.getBytes("UTF-8");
 		} catch (UnsupportedEncodingException e) {
 			// Handle exception
 		}
 
+		try {
+			PackageManager packageManager = this.mContext.getPackageManager();
+			PackageInfo info = packageManager.getPackageInfo("com.zebra.printconnect", 0);
+			long versionCode = info.getLongVersionCode();
+			if (versionCode >= NEW_PRINT_CONNECT_VERSION) {
+				requiresForeground = true;
+			}
+		} catch (PackageManager.NameNotFoundException e) {
+			callbackContext.error(e.getMessage());
+		}
```

Comment:
Not sure how Java works but could we make the catch more generic?

Reply:
Java is more strict about its exceptions -- `getPackageInfo` is declared as throwing a `PackageManager.NameNotFoundException`, so that is (essentially) the only exception it can throw. Similarly, because we call that function, we must either catch it (as here) or declare that this calling function may also throw the exception.

I think there's some nuance that means that undeclared exceptions could be thrown in specific circumstances, but there isn't the same sort of benefit to broad exception catches as there is in a interpreted language like PHP.

Reply:
Okay that's good to know. I think you did the right thing